src/callrpc.cpp

```diff
@@ -189,8 +189,11 @@ bool IsConfirmedBitcoinBlock(const uint256& hash, int nMinConfirmationDepth)
         UniValue result = find_value(reply, "result");
         if (!result.isObject())
             return false;
-        result = find_value(result.get_obj(), "confirmations");
-        return result.isNum() && result.get_int64() >= nMinConfirmationDepth;
+
+        UniValue confirmations = find_value(result.get_obj(), "confirmations");
+        UniValue nTx = find_value(result.get_obj(), "nTx");
+        return confirmations.isNum() && confirmations.get_int64() >= nMinConfirmationDepth
+            && (nbTxs == 0 || nTx.get_int64() == nbTxs);
```

Comment:
it might be pretty interesting to actually print a relevant line for this part failing. Would be nice to record an actual attack.